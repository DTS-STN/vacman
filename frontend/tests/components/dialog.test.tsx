import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import {
  Dialog,
  DialogClose,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from '~/components/dialog';

describe('Dialog', () => {
  it('renders Dialog component', () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>Open Dialog</DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Dialog Title</DialogTitle>
            <DialogDescription>Dialog Description</DialogDescription>
          </DialogHeader>
          <DialogFooter>
            <DialogClose>Close Dialog</DialogClose>
          </DialogFooter>
        </DialogContent>
      </Dialog>,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('renders DialogOverlay with correct class name', () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>Open Dialog</DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Dialog Title</DialogTitle>
          </DialogHeader>
        </DialogContent>
      </Dialog>,
    );
    expect(container).toMatchSnapshot('expected html');
  });

  it('renders DialogTitle and DialogDescription', () => {
    const { container } = render(
      <Dialog>
        <DialogTrigger>Open Dialog</DialogTrigger>
        <DialogContent>
          <DialogHeader>
            <DialogTitle>Dialog Title</DialogTitle>
            <DialogDescription>Dialog Description</DialogDescription>
          </DialogHeader>
        </DialogContent>
      </Dialog>,
    );
    expect(container).toMatchSnapshot('expected html');
  });
});
