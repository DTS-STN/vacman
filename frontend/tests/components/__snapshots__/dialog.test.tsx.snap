// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog > renders Dialog component > expected html 1`] = `
<div>
  <button
    aria-controls="radix-«r0»"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    Open Dialog
  </button>
</div>
`;

exports[`Dialog > renders DialogOverlay with correct class name > expected html 1`] = `
<div>
  <button
    aria-controls="radix-«r3»"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    Open Dialog
  </button>
</div>
`;

exports[`Dialog > renders DialogTitle and DialogDescription > expected html 1`] = `
<div>
  <button
    aria-controls="radix-«r6»"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    Open Dialog
  </button>
</div>
`;
